build/dangerjs/detectNonApprovedDependencies/utils/getRuntimeDependencies.ts

```diff
@@ -0,0 +1,33 @@
+import { spawnSync } from 'child_process'
+
+import config from '../../../../config'
+
+const { paths } = config
+
+const getRuntimeDependencies = (stardustPackageName: string) => {
+  const dependencyRegex = /^dependency:\s+(.*)$/
+  const result = spawnSync(
+    `yarn gulp test:dependencies:list --prefix='dependency: ' --package=${stardustPackageName}`,
+    {
+      shell: true,
+      cwd: paths.base(),
+      stdio: 'pipe',
+      encoding: 'utf-8',
+    },
+  )
+
+  const output = `${result.stdout}`
+  const error = `${result.stderr}`
+
+  if (error) {
+    throw new Error(error)
+  }
+
+  return output
+    .split('\n')
+    .map(line => line.match(dependencyRegex))
+    .filter(match => !!match)
+    .map(match => match[1])
```

Comment:
This thing looks like a bit overcomplicated to me. To get dependencies we need to run `webpack` with `license-webpack-plugin`, instead of this:
1) Gun Gulp task
2) Run Webpack
3) Parse task output

So, may be we can inline this check into this function?

Reply:
It also fails on Windows...

```
[11:12:01] Task never defined: '
[11:12:01] To list available tasks, try running: gulp --tasks
error Command failed with exit code 1.

Error: [11:12:01] Task never defined: '
[11:12:01] To list available tasks, try running: gulp --tasks
error Command failed with exit code 1.

    at Object.getRuntimeDependencies (C:\Users\olfedias\WebstormProjects\react\build
\dangerjs\detectNonApprovedDependencies\utils\getRuntimeDependencies.ts:17:15)
    at Object.<anonymous> (C:\Users\olfedias\WebstormProjects\react\build\dangerjs\d
etectNonApprovedDependencies\index.ts:57:48)
    at step (C:\Users\olfedias\WebstormProjects\react\build\dangerjs\detectNonApprov
edDependencies\index.ts:32:23)
    at Object.next (C:\Users\olfedias\WebstormProjects\react\build\dangerjs\detectNo
nApprovedDependencies\index.ts:13:53)
    at C:\Users\olfedias\WebstormProjects\react\build\dangerjs\detectNonApprovedDepe
ndencies\index.ts:7:71
    at new Promise (<anonymous>)
    at __awaiter (C:\Users\olfedias\WebstormProjects\react\build\dangerjs\detectNonA
pprovedDependencies\index.ts:3:12)
    at Object.detectNonApprovedDependencies [as default] (C:\Users\olfedias\Webstorm
Projects\react\build\dangerjs\detectNonApprovedDependencies\index.ts:50:66)
    at Object.<anonymous> (dangerfile.ts:45:81)
    at step (dangerfile.ts:32:23)
```

Reply:
for the first part of the question - unfortunately, this is a necessary complication, as otherwise there will be a problem for `license-webpack-plugin` to crawl through the imported modules. Have added comment that will describe this.

Have debugged Windows-related issue with @layershifter, it was caused by double quotes in the string command. This issue was fixed